community/kernel/src/main/java/org/neo4j/kernel/impl/index/GBPTreeUtil.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) 2002-2017 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.impl.index;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.NoSuchFileException;
+
+import org.neo4j.index.internal.gbptree.GBPTree;
+import org.neo4j.io.fs.FileHandle;
+import org.neo4j.io.pagecache.PageCache;
+
+/**
+ * Utilities for common operations around a {@link GBPTree}.
+ */
+public class GBPTreeUtil
+{
+    /**
+     * Deletes store file backing a {@link GBPTree}.
+     *
+     * @param pageCache {@link PageCache} which manages the file.
+     * @param storeFile the {@link File} to delete.
+     * @throws NoSuchFileException if the {@code storeFile} doesn't exist according to the {@code pageCache}.
+     * @throws IOException on failure to delete existing {@code storeFile}.
+     */
+    public static void delete( PageCache pageCache, File storeFile ) throws IOException
+    {
+        FileHandle fileHandle = storeFileHandle( pageCache, storeFile );
+        fileHandle.delete();
+    }
+
+    /**
+     * Deletes store file backing a {@link GBPTree}, if it exists according to the {@code pageCache}.
+     *
+     * @param pageCache {@link PageCache} which manages the file.
+     * @param storeFile the {@link File} to delete.
+     * @throws IOException on failure to delete existing {@code storeFile}.
+     */
+    public static void deleteIfPresent( PageCache pageCache, File storeFile ) throws IOException
```

Comment:
It could still fail on actually deleting the file. It's good to have it strict in some cases, others not.